### PR TITLE
Feature update auth check

### PR DESCRIPTION
**Pull request by blackbaud-brandonhare**

### CODE REVIEW

File: src/auth-email-whitelist.js [hunk 1]
```diff
@@ -15,22 +15,8 @@ const decode = require('jwt-decode');
...
-        'microedge.com',
-        'attentive.ly',
-        'everydayhero.com',
-        'smarttuition.com'
-      ];
       const parsedToken = decode(token);
-      const domain = parsedToken.email.split('@')[1];
-
-      if (emailDomainWhitelist.indexOf(domain.toLowerCase()) > -1) {
+      if (parsedToken['1bb.perms'] === 1 || parsedToken['1bb.perms'].indexOf(1) > -1) {
```
Comment (Blackbaud-BobbyEarl) [inline src/auth-email-whitelist.js]: For my clarification... `1bb.perms` can return a single integer or an array of integers?

Comment (blackbaud-brandonhare) [reply src/auth-email-whitelist.js]: Yep. It's a quirk in the .net side of things apparently.

Comment (Blackbaud-BobbyEarl) [reply src/auth-email-whitelist.js]: I believe checking this way will produce an error if `1bb.perms` is a single integer that's not `1`.

Comment (blackbaud-brandonhare) [reply src/auth-email-whitelist.js]: Good catch. Fixed.

**Comment by Blackbaud-SteveBrush** [comment]
@blackbaud-brandonhare: Still working on the unit tests?

**Comment by blackbaud-brandonhare** [comment]
@Blackbaud-SteveBrush Unit tests don't need to change. They're still applicable. We don't test in here that the plugin works.

**Review by Blackbaud-BobbyEarl — APPROVED** [review]
Thanks for adding that second check @blackbaud-brandonhare.  Functionality wise, I'm okay with this.  Personal preference and code readability wise, I think you might be able to rewrite that long if statement.  

Maybe something like one of the following, also accounting for the permission changing or there being more than one permission you need to check for.

```
if (typeof parsedToken['1bb.perms'] === 'number') {
  parsedToken['1bb.perms'] = [parsedToken['1bb.perms']];
}

return parsedToken['1bb.perms'].indexOf(1);
```

```
const requiredPerm = 1;
let hasRequiredPerm = false;

switch (typeof parsedToken['1bb.perms']) {
  case 'number':
    hasRequiredPerm = parsedToken['1bb.perms'] === requiredPerm;

  case 'Array';
    hasRequiredPerm = parsedToken['1bb.perms'].indexOf(requiredPerm);
}

if (hasRequiredPerm) {
  return Promise.resolve(true);
}
```

**Comment by Blackbaud-SteveBrush** [comment]
I agree with @Blackbaud-BobbyEarl. If you could rewrite the `if` statement in a more readable way, it would make future debugging that much easier.

**Review by Blackbaud-SteveBrush — APPROVED** [review]

**Pull request closed without merging**